From 5a20c7494f48da93914f50bbb54423ef540ef998 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Vr=C3=A1til?= <dvratil@kde.org>
Date: Thu, 27 Apr 2017 17:24:03 +0200
Subject: [PATCH 2/3] Auth: don't store cookies

Don't store WebEngine cookies, otherwise Google will display the account
that user signed in with the last time, which may not be desirable.
---
 src/core/ui/authwidget_p.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/core/ui/authwidget_p.cpp b/src/core/ui/authwidget_p.cpp
index 75d38cd..a51f4a9 100644
--- a/src/core/ui/authwidget_p.cpp
+++ b/src/core/ui/authwidget_p.cpp
@@ -26,6 +26,7 @@
 #include "../../debug.h"
 
 #include <QWebEngineView>
+#include <QWebEngineProfile>
 #include <QNetworkReply>
 #include <QContextMenuEvent>
 
@@ -36,6 +37,9 @@ using namespace KGAPI2;
 WebView::WebView(QWidget *parent)
     : QWebEngineView(parent)
 {
+    // Don't store cookies, so that subsequent invocations of AuthJob won't remember
+    // the previous accounts.
+    QWebEngineProfile::defaultProfile()->setPersistentCookiesPolicy(QWebEngineProfile::NoPersistentCookies);
 }
 
 
-- 
2.12.2

